lib/consumer_supervisor.ex

```diff
@@ -262,6 +258,23 @@ defmodule ConsumerSupervisor do
     call(supervisor, :count_children)
   end
 
+  @doc """
+  Receives a template to initialize and a set of options.
+
+  This is typically invoked at the end of the init/1 callback of module-based supervisors.
+
+  This function returns a the child specification and the supervisor flags.
+
+  """
+  def init({_, _, _, _, _, _} = template, opts) do
```

Comment:
I decided to only support one `template` and not a list of it which can only contain one value, since this whole supervisor only supports one template.

If you do not like that I can change it to support a list.